### PR TITLE
Updating whatsnew for 2.4.0-develop

### PR DESCRIPTION
**Pull request by erikmarr**
## Purpose of this pull request

This pull request (PR) adds whatsnew entries for 2.4.0 features.

## Affected DevDocs pages

-  https://devdocs.magento.com/whats-new.html

### CODE REVIEW

**Comment by jeff-matthews** [comment]
Thanks @erikmarr! Jeanne is about to merge the release notes PR #7301, so please add that too. That's the last PR we'll need though.

**Comment by jeff-matthews** [comment]
@erikmarr, it's been merged.

File: src/_data/whats-new.yml [hunk 1]
```diff
@@ -4,8 +4,102 @@ description:
...
-updated: Thu May 21 16:26:01 2020
+updated: Thu May 28 13:20:08 2020
 entries:
+- description: Added a new section about whitelisting inline scripts and styles in
+    a [content security policy](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/security/content-security-policies.html#whitelist-an-inline-script-or-style).
+  versions: 2.4.0
+  type: Major Update
+  date: May 28, 2020
+  link: https://github.com/magento/devdocs/pull/7293
+- description: Added a new section to [MySQL](https://devdocs.magento.com/guides/v2.3/install-gde/prereq/mysql.html)
```
Comment (jeff-matthews) [inline src/_data/whats-new.yml:21]: The link in #7293 should point to 2.4 docs instead of 2.3 I think.

File: src/_data/whats-new.yml [hunk 2]
```diff
@@ -4,8 +4,102 @@ description:
...
+- description: Added BIC [highlights](https://devdocs.magento.com/guides/v2.4/release-notes/backward-incompatible-changes/index.html)
+    and [reference]( https://devdocs.magento.com/guides/v2.4/release-notes/backward-incompatible-changes/reference.html)
+    to support the 2.4.0 beta release.
+  versions: 2.4.0
+  type: Major Update
+  date: May 28, 2020
+  link: https://github.com/magento/devdocs/pull/7295
+- description: Removed references to the Web Setup Wizard, Component Manager, Module
+    Manager, System Upgrade Utility, and Magento "updater" application because the
+    functionality was removed in 2.4.0. Updated the [installation flow diagram](https://devdocs.magento.com/guides/v2.3/install-gde/install-flow-diagram.html)
```
Comment (jeff-matthews) [inline src/_data/whats-new.yml:47]: Links in PR should point to 2.4 docs. That's my PR so I'll fix the PR.

File: src/_data/whats-new.yml [hunk 3]
```diff
@@ -4,8 +4,102 @@ description:
...
+    and optimized the [Software Update Guide](https://devdocs.magento.com/guides/v2.3/comp-mgr/bk-compman-upgrade-guide.html).
+  versions: 2.4.0
+  type: Major Update
+  date: May 28, 2020
+  link: https://github.com/magento/devdocs/pull/7287
+- description: Removed the option to disable 2FA from the [Two-Factor Authentication<br/>](https://devdocs.magento.com/guides/v2.4/security/two-factor-authentication.html)
```
Comment (jeff-matthews) [inline src/_data/whats-new.yml:53]: The description has a rogue `<br/>` tag that needs to be removed.

File: src/_data/whats-new.yml [hunk 4]
```diff
@@ -4,8 +4,102 @@ description:
...
+    topic and added a new section describing the Magento Web API workflow.
+  versions: 2.4.0
+  type: Major Update
+  date: May 28, 2020
+  link: https://github.com/magento/devdocs/pull/7262
+- description: Removed the topic about using the custom script to upgrade Magento.
+    The [Composer update plugin](https://devdocs.magento.com/guides/v2.3/comp-mgr/cli/upgrade-with-plugin.html)
```
Comment (jeff-matthews) [inline src/_data/whats-new.yml:60]: Link to 2.4 instead of 2.4. I'll fix the Pr.

**Comment by erikmarr** [comment]
@jeff-matthews on it!

File: src/_data/whats-new.yml [hunk 5]
```diff
@@ -4,8 +4,102 @@ description:
...
+    replaces the custom script.
+  versions: 2.3.x, 2.4.0
+  type: Major Update
+  date: May 28, 2020
+  link: https://github.com/magento/devdocs/pull/7232
+- description: Removed beta information from the [Composer root update plugin](https://devdocs.magento.com/guides/v2.3/comp-mgr/cli/upgrade-with-plugin.html)
```
Comment (jeff-matthews) [inline src/_data/whats-new.yml:66]: links again...

File: src/_data/whats-new.yml (hunk 1 above)
Comment (jeff-matthews) [reply src/_data/whats-new.yml:21]: Fixed PR description

File: src/_data/whats-new.yml (hunk 2 above)
Comment (jeff-matthews) [reply src/_data/whats-new.yml:47]: Fixed PR description

File: src/_data/whats-new.yml (hunk 3 above)
Comment (jeff-matthews) [reply src/_data/whats-new.yml:53]: Fixed PR description

File: src/_data/whats-new.yml (hunk 4 above)
Comment (jeff-matthews) [reply src/_data/whats-new.yml:60]: Fixed PR description

File: src/_data/whats-new.yml (hunk 5 above)
Comment (jeff-matthews) [reply src/_data/whats-new.yml:66]: Fixed PR description

File: src/_data/whats-new.yml (hunk 3 above)
Comment (jeff-matthews) [reply src/_data/whats-new.yml:53]: Fixed description

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
running tests

**Pull request merged**